whitepaper.tex

```diff
@@ -765,115 +809,146 @@
-  automatically translates CAmkES ADL into CapDL, indicated by the box
+  representation. The Microkit framework contains a compiler which
+  automatically translates the SDF into CapDL, indicated by the box
   arrow pointing left-down. The box in the left of the figure gives a
   (simplified) representation of the seL4 objects described in
   CapDL. (It is actually a simplified representation of a much simpler
-  system, basically just the two components at the top of
-  \autoref{f:camkes} and the connector between them.)
+  system, basically just two of the PDs at the top of
+  \autoref{f:microkit} and the channel between them.)
 
   The CapDL spec is a precise representation of access rights in the
   system, and it is what seL4 enforces. Which means that once the
   system gets into the state described by the CapDL spec, it is
-  guaranteed to behave as described by the CAmkES ADL spec, and
+  guaranteed to behave as described by the SDF spec, and
   therefore the architecture-level description is sufficient for
   further reasoning about security properties.
 
   So we need assurance that the system boots up into the state
   described by the CapDL spec. We achieve this with a second
   automated step: We generate from CapDL the startup code that, as
   soon as seL4 itself has booted, takes control and generates all the
-  seL4 objects referenced by the spec, including the ones describing
-  active components, and distributes the \emph{capabilities} (see
-  \autoref{s:caps}) that   grant access to those objects according to
+  seL4 objects referenced by the spec, including the ones representing
+  PDs, and distributes the \emph{capabilities} (see
+  \autoref{s:caps}) that grant access to those objects according to
   the spec. At the end of the execution of this init code, the system is
   provably in the state described by the CapDL spec, and thus in the
-  state represented by the ADL spec.
-
-  The third thing that gets generated from the ADL spec is the
-  ``glue'' code between components. Sending data through a
-  connector requires invocation of seL4 system calls, the exact
-  details of which are hidden by the CAmkES abstraction. The glue code
-  is   setting up these system  calls. For example, an ``RPC''
-  connector abstracts the invocation of a function provided by another
-  component as a regular function call performed by the client
-  component.
-\iffalse
-  The glue-code generation is also verified, to guarantee
-  that the invocation across component boundaries is functionally
-  equivalent to executing the function's code (and using its state)
-  inside the client component (while enforcing the module interface).
-\fi
+  state represented by the SDF spec.
 
-  \textbf{Note:} At the time of writing, the proofs about CAmkES and
+  \code{libmicrokit}, which maps the Microkit abstractions to the seL4 API,
+  also verified, which guarantees that seL4 system calls are invoked correctly.
```

Comment:
Do we want to make this claim? The latest released version of Microkit that people are using has diverged from what the proofs originally worked on and have not been caught up yet.

Reply:
fair point. In fact, both this and the CapDL mapping verification are about as incomplete as the CAmkES proofs. The note immediately following does introduce the caveat. Is that not enough? I can add that they are based on an older version.

Reply:
How about changing the note to read:

> At the time of writing, the proofs about Microkit and
  CapDL are not yet complete and based on an older version of the
  Microkit.  Completion is planned for the future.

Reply:
As far as I understood the proofs, they do not actually show that they invoke seL4 correctly in any case -- for that you'd need a connection to the seL4 spec. Isn't that what Rob is currently working on?

Reply:
Yes, they use a manually-derived spec. Which is part of being "incomplete". But I guess it's better not to be more conservative, and just say "verified to be implemented correctly"?